### PR TITLE
glaze: add version 1.9.5, remove older versions, relax gcc version check

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **glaze/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by RazielXYZ** [comment]
According to the release note for 1.9.0 (https://github.com/stephenberry/glaze/releases/tag/v1.9.0) gcc 11 should still be supported; any idea why it's not working?

**Comment by toge** [comment]
@RazielXYZ 
Thank you for your comment.
I don't know the reason not to work on gcc11 in cci environment.

A workaround is use `NOT_ON_C3I` environment which is used by `qt/6.x.x` recipes.
https://github.com/conan-io/conan-center-index/blob/master/recipes/qt/6.x.x/conanfile.py#L235-L239

I am going to apply these idea later.

**Review by RazielXYZ — APPROVED** [review]

File: recipes/glaze/all/conanfile.py [hunk 1]
```diff
@@ -40,9 +40,18 @@ def package_id(self):
...
+        # there is a compilation error on cci's gcc 11.
+        # https://c3i.jfrog.io/c3i/misc/logs/pr/21629/1-linux-gcc/glaze/1.9.0//5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9-test.txt
+        if os.getenv('NOT_ON_C3I', '0') == '0':
+            if Version(self.version) >= "1.9.0" and \
+                self.info.settings.compiler == "gcc" and \
+                Version(self.info.settings.compiler.version) == "11":
+                raise ConanInvalidConfiguration(f"gcc 11 doesn't compile {self.ref} in C3I due to internal compiler error.\n" \
+                                                "However, it works fine with gcc11 in several environments.\n"
+                                                f"If you want to try {self.ref} with gcc11, set environment variable NOT_ON_C3I=1.")
+
```
Comment (uilianries) [inline recipes/glaze/all/conanfile.py:52]:
```suggestion
```

- The log in this line points to the version 1.9.0, which is removed in this pull request. So we no longer build it.

- I built locally 1.9.5 with GCC 11 and it works:

```
$ conan create . --version=1.9.5 

======== Exporting recipe to the cache ========
glaze/1.9.5: Exporting package recipe: /home/conan/project/conanfile.py
glaze/1.9.5: exports: File 'conandata.yml' found. Exporting it...
glaze/1.9.5: Copied 1 '.yml' file: conandata.yml
glaze/1.9.5: Copied 1 '.py' file: conanfile.py
glaze/1.9.5: Exported to cache folder: /home/conan/.conan2/p/glaze5276c32c7057e/e
glaze/1.9.5: Exported: glaze/1.9.5#0e63d7dd896dce2db0e269621cae3a1b (2023-12-12 08:47:24 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux


======== Computing dependency graph ========
Graph root
    cli
Requirements
    glaze/1.9.5#0e63d7dd896dce2db0e269621cae3a1b - Cache

======== Computing necessary packages ========
glaze/1.9.5: Forced build from source
Requirements
    glaze/1.9.5#0e63d7dd896dce2db0e269621cae3a1b:da39a3ee5e6b4b0d3255bfef95601890afd80709 - Build

======== Installing packages ========

-------- Installing package glaze/1.9.5 (1 of 1) --------
glaze/1.9.5: Building from source
glaze/1.9.5: Package glaze/1.9.5:da39a3ee5e6b4b0d3255bfef95601890afd80709
glaze/1.9.5: Building your package in /home/conan/.conan2/p/b/glaze80a5461f2eb08/b
glaze/1.9.5: Generating aggregated env files
glaze/1.9.5: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
glaze/1.9.5: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' built
glaze/1.9.5: Build folder /home/conan/.conan2/p/b/glaze80a5461f2eb08/b/build-release
glaze/1.9.5: Generating the package
glaze/1.9.5: Packaging in folder /home/conan/.conan2/p/b/glaze80a5461f2eb08/p
glaze/1.9.5: Calling package()
glaze/1.9.5: package(): Packaged 1 file: LICENSE
glaze/1.9.5: package(): Packaged 98 '.hpp' files
glaze/1.9.5: Created package revision e4b36628448a64cb0a4ae9f8d964948b
glaze/1.9.5: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created
glaze/1.9.5: Full package reference: glaze/1.9.5#0e63d7dd896dce2db0e269621cae3a1b:da39a3ee5e6b4b0d3255bfef95601890afd80709#e4b36628448a64cb0a4ae9f8d964948b
glaze/1.9.5: Package folder /home/conan/.conan2/p/b/glaze21e366b9bdd85/p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    glaze/1.9.5 (test package): /home/conan/project/test_package/conanfile.py
Requirements
    glaze/1.9.5#0e63d7dd896dce2db0e269621cae3a1b - Cache

======== Computing necessary packages ========
Requirements
    glaze/1.9.5#0e63d7dd896dce2db0e269621cae3a1b:da39a3ee5e6b4b0d3255bfef95601890afd80709#e4b36628448a64cb0a4ae9f8d964948b - Cache

======== Installing packages ========
glaze/1.9.5: Already installed! (1 of 1)

======== Testing the package ========
Removing previously existing 'test_package' build folder: /home/conan/project/test_package/build/gcc-11-x86_64-20-release
glaze/1.9.5 (test package): Test package build: build/gcc-11-x86_64-20-release
glaze/1.9.5 (test package): Test package build folder: /home/conan/project/test_package/build/gcc-11-x86_64-20-release
glaze/1.9.5 (test package): Writing generators to /home/conan/project/test_package/build/gcc-11-x86_64-20-release/generators
glaze/1.9.5 (test package): Generator 'CMakeToolchain' calling 'generate()'
glaze/1.9.5 (test package): CMakeToolchain generated: conan_toolchain.cmake
glaze/1.9.5 (test package): CMakeToolchain generated: CMakePresets.json
glaze/1.9.5 (test package): CMakeToolchain generated: ../../../CMakeUserPresets.json
glaze/1.9.5 (test package): Generator 'CMakeDeps' calling 'generate()'
glaze/1.9.5 (test package): Generator 'VirtualRunEnv' calling 'generate()'
glaze/1.9.5 (test package): Generating aggregated env files
glaze/1.9.5 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
glaze/1.9.5 (test package): Calling build()
glaze/1.9.5 (test package): Running CMake.configure()
glaze/1.9.5 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/conan/project/test_package/build/gcc-11-x86_64-20-release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/conan/project/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/conan/project/test_package"
-- Using Conan toolchain: /home/conan/project/test_package/build/gcc-11-x86_64-20-release/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- The CXX compiler identification is GNU 11.1.0
-- Check for working CXX compiler: /usr/local/bin/c++
-- Check for working CXX compiler: /usr/local/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'glaze::glaze'
-- Configuring done
-- Generating done
-- Build files have been written to: /home/conan/project/test_package/build/gcc-11-x86_64-20-release

glaze/1.9.5 (test package): Running CMake.build()
glaze/1.9.5 (test package): RUN: cmake --build "/home/conan/project/test_package/build/gcc-11-x86_64-20-release" -- -j10
Scanning dependencies of target test_package
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[100%] Linking CXX executable test_package
[100%] Built target test_package


======== Testing the package: Executing test ========
glaze/1.9.5 (test package): Running test()
glaze/1.9.5 (test package): RUN: ./test_package
```

**Comment by toge** [comment]
@uilianries 
Thank you for your suggestion.
I could build locally 1.9.0 with GCC 11.
Let's try again.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**